sysbuild/mcuboot.cmake

```diff
@@ -5,10 +5,24 @@
 
 if(SB_CONFIG_MCUBOOT_BUILD_DIRECT_XIP_VARIANT)
   set(image mcuboot_secondary_app)
-  ExternalNcsVariantProject_Add(APPLICATION ${DEFAULT_IMAGE} VARIANT ${image})
+  if(SB_CONFIG_PARTITION_MANAGER)
+    ExternalNcsVariantProject_Add(APPLICATION ${DEFAULT_IMAGE} VARIANT ${image})
 
-  set_property(GLOBAL APPEND PROPERTY
-      PM_APP_IMAGES
-      "${image}"
-  )
+    set_property(GLOBAL APPEND PROPERTY
+        PM_APP_IMAGES
+        "${image}"
+    )
+  else()
+    # TODO: NCSDK-33774 add a general way to pass configuration options to the variant
+    # image
+    zephyr_get(extra_conf_file SYSBUILD LOCAL VAR EXTRA_CONF_FILE)
+
+    ExternalNcsVariantProject_Add(APPLICATION ${DEFAULT_IMAGE} VARIANT ${image} SPLIT_KCONFIG true)
```

Comment:
this looks like it has more holes than a leaking bucket in terms of things is misses @tejlmand for discussion and analysis